### ISSUE TITLE
Reduce Swift warnings

### ISSUE THREAD

**Issue by notmandatory** [mozilla/uniffi-rs#1460]
Due to #1350 which escapes all Swift identifiers (except for namespace functions) users of generated Swift bindings now get the follow warnings: 

```
Keyword '<name>' does not need to be escaped in argument list
```

There's currently no way in Swift to suppress these warnings and I was unable to find any active proposals to add a way to do it.

One way to avoid these is to only escape known Swift keywords. The best list of keywords I've found is in the blog post: ["All the Reserved Keywords in Swift"](https://betterprogramming.pub/all-the-reserved-keywords-in-swift-17efcfaa3f3e). 

I'm opening this issue for feedback on if this is worth fixing and as a starting point for myself or anyone else who wants to do it.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/UNIFFI-234)
┆Issue Number: UNIFFI-234

**Comment by mhammond** [comment]
The warnings annoy me too, but not enough to actually make the change :) I think having a list of keywords would be fine - [that's what Python does](https://github.com/mozilla/uniffi-rs/blob/main/uniffi_bindgen/src/bindings/python/gen_python/mod.rs#L30)

**Comment by notmandatory** [comment]
Thanks for the link, good to see hard-coding a word list isn't too hack-y a way to handle it.

**Comment by badboy** [comment]
I'm still mad that there's no way in Swift to disable warnings. Wargh.

**Comment by mhammond** [comment]
yeah, I dug into this looking for an easy answer a little while ago - but nope :(

**Comment by ijc** [comment]
Between #1701, #1703 and #1704 there are no more Swift warnings in the Uniffi test fixtures.

Specifically #1701 addresses the issue mention here regarding unnecessary quoting of non-keywords.
